### ISSUE TITLE
Crash when attempting to reweight a CCQE-strange event

### ISSUE THREAD

**Issue by sjgardiner** [GENIE-MC/Reweight#12]
A MicroBooNE user reports a fatal error when running the MaCCQE weight calculator over a sample containing a CCQE-strange event. The error message was
`1588038661 FATAL KineLimits : [s] <Jacobian (255)> : *** Can not compute Jacobian for transforming: <QELEvGen> --> <{Q2}|E>`

**Comment by sjgardiner** [comment]
Here is a printout of the full GENIE event record that caused the error:
```
|------------------------------------------------------------------------------------------------------------------|
|GENIE GHEP Event Record [print level:  13]                                                                        |
|------------------------------------------------------------------------------------------------------------------|
| Idx |          Name | Ist |        PDG |   Mother  | Daughter  |    Px(x) |   Py(y) |   Pz(z) |    E(t) |     m  | 
|------------------------------------------------------------------------------------------------------------------|
|   0 |     nu_mu_bar |   0 |        -14 |  -1 |  -1 |   4 |   4 |   1.353 |   0.145 |   2.610 |   2.944 |   0.000 | 
|     |               |     |            |     |     |     |     |   4.163 |   3.706 |  -0.856 |   0.000 |         | 
|   1 |          Ar40 |   0 | 1000180400 |  -1 |  -1 |   2 |   3 |   0.000 |   0.000 |   0.000 |  37.216 |  37.216 | 
|     |               |     |            |     |     |     |     |   0.000 |   0.000 |   0.000 |   0.000 |         | 
|   2 |        proton |  11 |       2212 |   1 |  -1 |   5 |   5 |   0.046 |  -0.003 |   0.019 |   0.926 | **0.938 | M = 0.924 
|     |               |     |            |     |     |     |     |   4.163 |   3.706 |  -0.856 |   0.000 |         | 
|   3 |          Cl39 |   2 | 1000170390 |   1 |  -1 |   7 |   7 |  -0.046 |   0.003 |  -0.019 |  36.290 |  36.290 | 
|     |               |     |            |     |     |     |     |   0.000 |   0.000 |   0.000 |   0.000 |         | 
|   4 |           mu+ |   1 |        -13 |   0 |  -1 |  -1 |  -1 |   1.626 |  -0.219 |   1.559 |   2.266 |   0.106 | P = (0.718,-0.097,0.689)
|     |               |     |            |     |     |     |     |   4.163 |   3.706 |  -0.856 |   0.000 |         | 
|   5 |       Lambda0 |  14 |       3122 |   2 |  -1 |   6 |   6 |  -0.227 |   0.361 |   1.070 |   1.603 |   1.116 | 
|     |               |     |            |     |     |     |     |   4.163 |   3.706 |  -0.856 |   0.000 |         | 
|   6 |       Lambda0 |   1 |       3122 |   5 |  -1 |  -1 |  -1 |  -0.227 |   0.361 |   1.070 |   1.603 |   1.116 | 
|     |               |     |            |     |     |     |     |   4.163 |   3.706 |  -0.856 |   0.000 |         | 
|   7 |      HadrBlob |  15 | 2000000002 |   3 |  -1 |  -1 |  -1 |  -0.046 |   0.003 |  -0.019 |  36.290 | **0.000 | M = 36.290 
|     |               |     |            |     |     |     |     |   0.000 |   0.000 |   0.000 |   0.000 |         | 
|------------------------------------------------------------------------------------------------------------------|
|       Fin-Init:                                                |  -0.000 |   0.000 |  -0.000 |   0.000 |         | 
|------------------------------------------------------------------------------------------------------------------|
|       Vertex:      nu_mu_bar @ (x =     1.72093 m, y =     0.71045 m, z =     7.95260 m, t =    4.027527e+01 s)  |
|------------------------------------------------------------------------------------------------------------------|
| Err flag [bits:15->0] : 0000000000000000    |  1st set:                                                     none | 
| Err mask [bits:15->0] : 1111111111111111    |  Is unphysical:    NO |   Accepted:   YES                          |
|------------------------------------------------------------------------------------------------------------------|
| sig(Ev) =       3.57686e-39 cm^2  | dsig(Q2;E)/dQ2 =          4.17898e-41 cm^2/GeV^2 | Weight =          1.00000 |
|------------------------------------------------------------------------------------------------------------------|
--------------------------------------------------------------------------------------------------------------
GENIE Interaction Summary
--------------------------------------------------------------------------------------------------------------
[-] [Init-State] 
 |--> probe        : PDG-code = -14 (nu_mu_bar)
 |--> nucl. target : Z = 18, A = 40, PDG-Code = 1000180400 (Ar40)
 |--> hit nucleon  : PDC-Code = 2212 (proton)
 |--> hit quark    : no set
 |--> probe 4P     : (E =     2.943785, Px =     1.353250, Py =     0.144902, Pz =     2.610285)
 |--> target 4P    : (E =    37.215526, Px =     0.000000, Py =     0.000000, Pz =     0.000000)
 |--> nucleon 4P   : (E =     0.925709, Px =     0.045865, Py =    -0.003238, Pz =     0.018774)
[-] [Process-Info]  
 |--> Interaction : Weak[CC]
 |--> Scattering  : QES
[-] [Kinematics]
 |--> *Selected* Bjorken x = 0.685886
 |--> *Selected* Inelasticity y = 0.237609
 |--> *Selected* Momentum transfer Q2 (>0) = 0.852180
 |--> *Selected* Hadronic invariant mass W = 1.115680
[-] [Exclusive Process Info] 
 |--> charm prod.  : false |--> strange prod.  : true - Strange hadron PDG-code = 3122 (Lambda0)
 |--> f/s nucleons : N(p) = 0 N(n) = 0
 |--> f/s pions    : N(pi^0) = 0 N(pi^+) = 0 N(pi^-) = 0
 |--> resonance    : [not set]
--------------------------------------------------------------------------------------------------------------
```

**Comment by sjgardiner** [comment]
Hard cuts setting the event weight to unity exist in the CCQE weight calculators to protect against CCQE-Charm events. However, there are no such cuts applied for CCQE-Lambda events. This appears to be the source of the problem. New cuts should be added to the relevant weight calculators.

**Comment by Ooohu** [comment]
Got the same error 

`1630205145 FATAL KineLimits : [s] <Jacobian (258)> : *** Can not compute Jacobian for transforming: <QELEvGen> --> <{Q2}|E>`

for CCQE-Sigma0 events:

```
|------------------------------------------------------------------------------------------------------------------|
|GENIE GHEP Event Record [print level:   3]                                                                        |
|------------------------------------------------------------------------------------------------------------------|
| Idx |          Name | Ist |        PDG |   Mother  | Daughter  |      Px |      Py |      Pz |       E |      m  | 
|------------------------------------------------------------------------------------------------------------------|
|   0 |     nu_mu_bar |   0 |        -14 |  -1 |  -1 |   4 |   4 |  -0.013 |  -0.017 |   1.747 |   1.747 |   0.000 | 
|   1 |          Ar40 |   0 | 1000180400 |  -1 |  -1 |   2 |   3 |   0.000 |   0.000 |   0.000 |  37.216 |  37.216 | 
|   2 |        proton |  11 |       2212 |   1 |  -1 |   5 |   5 |   0.039 |  -0.156 |   0.027 |   0.925 | **0.938 | M = 0.911 
|   3 |          Cl39 |   2 | 1000170390 |   1 |  -1 |   7 |   7 |  -0.039 |   0.156 |  -0.027 |  36.290 |  36.290 | 
|   4 |           mu+ |   1 |        -13 |   0 |  -1 |  -1 |  -1 |   0.143 |  -0.633 |   0.922 |   1.132 |   0.106 | P = (0.127,-0.561,0.818)
|   5 |        Sigma0 |  14 |       3212 |   2 |  -1 |   6 |   6 |  -0.116 |   0.460 |   0.852 |   1.541 |   1.193 | 
|   6 |        Sigma0 |   1 |       3212 |   5 |  -1 |  -1 |  -1 |  -0.116 |   0.460 |   0.852 |   1.541 |   1.193 | 
|   7 |      HadrBlob |  15 | 2000000002 |   3 |  -1 |  -1 |  -1 |  -0.039 |   0.156 |  -0.027 |  36.290 | **0.000 | M = 36.290 
|------------------------------------------------------------------------------------------------------------------|
|       Fin-Init:                                                |   0.000 |   0.000 |  -0.000 |   0.000 |         | 
|------------------------------------------------------------------------------------------------------------------|
|       Vertex:      nu_mu_bar @ (x =    -1.85325 m, y =    -1.08519 m, z =     2.17111 m, t =    4.060004e-08 s)  |
|------------------------------------------------------------------------------------------------------------------|
| Err flag [bits:15->0] : 0000000000000000    |  1st set:                                                     none | 
| Err mask [bits:15->0] : 1111111111111111    |  Is unphysical:    NO |   Accepted:   YES                          |
|------------------------------------------------------------------------------------------------------------------|
| sig(Ev) =       1.25786e-39 cm^2  | dsig(Q2;E)/dQ2 =          1.76890e-41 cm^2/GeV^2 | Weight =          1.00000 |
|------------------------------------------------------------------------------------------------------------------|

--------------------------------------------------------------------------------------------------------------
GENIE Interaction Summary
--------------------------------------------------------------------------------------------------------------
[-] [Init-State] 
 |--> probe        : PDG-code = -14 (nu_mu_bar)
 |--> nucl. target : Z = 18, A = 40, PDG-Code = 1000180400 (Ar40)
 |--> hit nucleon  : PDC-Code = 2212 (proton)
 |--> hit quark    : no set
 |--> probe 4P     : (E =     1.747480, Px =    -0.012882, Py =    -0.016850, Pz =     1.747351)
 |--> target 4P    : (E =    37.215526, Px =     0.000000, Py =     0.000000, Pz =     0.000000)
 |--> nucleon 4P   : (E =     0.925375, Px =     0.039038, Py =    -0.156385, Pz =     0.027073)
[-] [Process-Info]  
 |--> Interaction : Weak[CC]
 |--> Scattering  : QES
[-] [Kinematics]
 |-->  *Running* lepton kinetic energy = 1.026588
 |-->  *Running* cosine of lepton theta = 0.817897
 |--> *Selected* Bjorken x = 0.543731
 |--> *Selected* Inelasticity y = 0.414231
 |--> *Selected* Momentum transfer Q2 (>0) = 0.706158
 |--> *Selected* Momentum transfer q2 (<0) = -0.706158
 |--> *Selected* Hadronic invariant mass W = 1.192550
[-] [Exclusive Process Info] 
 |--> charm prod.  : false |--> strange prod.  : true - Strange hadron PDG-code = 3212 (Sigma0)
 |--> f/s nucleons : N(p) = 0 N(n) = 0
 |--> f/s pions    : N(pi^0) = 0 N(pi^+) = 0 N(pi^-) = 0
 |--> resonance    : [not set]
--------------------------------------------------------------------------------------------------------------
```

**Comment by sjgardiner** [comment]
This problem will be resolved when pull request #18 is merged.
